### DISCUSSION TITLE
Help with DOI Resolver Plugin

### DISCUSSION THREAD

**Discussion by lost-geographer** [searx/searx#2950] [Installation issue: custom] [answered]
Hello,

Did someone managed to use the DOI rewriting plugin ? I tried to use it on different public instances (and on my own) but it doesn't seems to work. For example I looked for the string "https://doi.org/10.1038/s41586-018-0819-2" (I know it is available on sci-hub) but I am not able to see any particular rewritten result. Can someone reproduce this behavior ?

(I posted the question on the custom install group, but it might be related to others)

Cheers !

**Comment by 0xhtml** [comment, marked as answer]
Be default results are rewritten to oadoi.org. Have you selected sci-hub as the DOI resolver in settings?

Searching for `!science A warped disk around an infant protostar` results in rewritten results to sci-hub for me. Rewritten results are coming from crossref and semantic scholar for that specific search. Which engines do you have enabled?

**Reply by lost-geographer** [reply]
Ok, thanks for your answer. Indeed I edited the resolver parameter in the settings.yml as follows: `default_doi_resolver : 'sci-hub.se'` and all the "science" engines are enabled... Actually I am talking about [this particular instance](https://search.lostgeographer.eu/), but I got the same issue on [this](https://searx.fmac.xyz/) and [that](https://searx.info/) instance as well.

I will try to run the instance in debug mode to see if the plug-in is correctly loaded.

**Reply by 0xhtml** [reply]
The problem seems to be with setting the default_doi_resolver. If I go to settings on https://search.lostgeographer.eu and select sci-hub.se (oadoi.org is selected by default) manually, the rewrite to sci-hub works.

Are you running a version later than https://github.com/searx/searx/commit/ebfd0eb2b79f73eef108ac1ad89c07174ad6441c?

**Reply by lost-geographer** [reply]
Nice catch @0xhtml ! Actually my version seems to be older than the `ebfd0eb` commit (it's a [repacked version](https://github.com/YunoHost-Apps/searx_ynh) by the yunohost team). I changed the files myself and it works !
